src/main/kotlin/com/ritense/iko/aggregateddataprofile/domain/AggregatedDataProfile.kt

```diff
@@ -100,7 +100,16 @@ class AggregatedDataProfile(
     }
 
     fun removeRelation(request: DeleteRelationForm) {
-        this.relations.removeIf { it.id == request.id }
+        // Remove the selected relation and all its descendants
+        val toRemove: MutableSet<UUID> = linkedSetOf()
+
+        fun collectDescendants(parentId: UUID) {
```

Comment:
Local functions this is new to me not saying its bad but why need it. Convince me please. Also deleting descendants is kinda scary losing data.... I mean.